src/EditorFeatures/Core/ExternalAccess/VSTypeScript/VSTypeScriptDiagnosticService.cs

```diff
@@ -33,18 +33,37 @@ public async Task<ImmutableArray<VSTypeScriptDiagnosticData>> GetPushDiagnostics
             return result.SelectAsArray(data => new VSTypeScriptDiagnosticData(data));
         }
 
+        [Obsolete]
         public IDisposable RegisterDiagnosticsUpdatedEventHandler(Action<VSTypeScriptDiagnosticsUpdatedArgsWrapper> action)
             => new EventHandlerWrapper(_service, action);
 
+        public IDisposable RegisterDiagnosticsUpdatedEventHandler(Action<ImmutableArray<VSTypeScriptDiagnosticsUpdatedArgsWrapper>> action)
+            => new EventHandlerWrapper(_service, action);
+
         private sealed class EventHandlerWrapper : IDisposable
         {
             private readonly IDiagnosticService _service;
-            private readonly EventHandler<DiagnosticsUpdatedArgs> _handler;
+            private readonly EventHandler<ImmutableArray<DiagnosticsUpdatedArgs>> _handler;
 
+            [Obsolete]
             internal EventHandlerWrapper(IDiagnosticService service, Action<VSTypeScriptDiagnosticsUpdatedArgsWrapper> action)
```

Comment:
>internal EventHandlerWrapper(IDiagnosticService service, Action<VSTypeScriptDiagnosticsUpdatedArgsWrapper> action) [](http://example.com/codeflow?start=12&length=114)

thoughts on changing this one to just call the other ctor?

Reply:
➡️ Leaving this one as-is